### DISCUSSION TITLE
When using docker-compose, if ports 443 or 80 are occupied, how to resolve it?General Discussion

### DISCUSSION THREAD

**Discussion by itboyhp** [langgenius/dify#6780] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

When using docker-compose, if ports 443 or 80 are occupied, how to resolve it?

**Comment by akramer-zibra** [comment]
@itboyhp you can override both published ports with environment variables: `EXPOSE_NGINX_PORT` and `EXPOSE_NGINX_SSL_PORT`

https://github.com/langgenius/dify/blob/56b43f62d1a7d677fa5d1200efbc43f0535a3f1a/docker/docker-compose.yaml#L357-L359

If you already use a `.env` file for configuration (which you may should), simply change both values to ports your system does not use yet. For example `8080` and `8433` (or `9080` and `9443` or ...). 

https://github.com/langgenius/dify/blob/56b43f62d1a7d677fa5d1200efbc43f0535a3f1a/docker/.env.example#L662-L666

Be aware that you will also have to adapt the service URLs in the configuration with the ports you have choosen aswell. For example: `CONSOLE_API_URL=http://...:8080`, `CONSOLE_WEB_URL=http://...:8080`, `SERVICE_API_URL=http://...:8080`, `APP_API_URL=http://....:8080` and `APP_WEB_URL=http://...:8080`. 

If you use dify in a local experimental setup you may use `http` protocol with port e.g. `8080`. The `https`-protocol setup requires some tls certificates, so nginx can exncrypt the traffic. This makes the setup more secure for production use, but also more complex.
